amida-tech/indaba-client
# Feature/INBA-473 Inbox thread

#### What's this PR do?
Make inbox entries threads instead of individual messages

#### Related JIRA tickets:
[DS-473](https://jira.amida-tech.com/browse/INBA-473)

#### How should this be manually tested?
1. Generate message threads consisting of several messages from different users
1. Check that each thread appears in the inbox or archive list once, with the number of messages in the thread indicated. Threads of only one message do not display the count
1. Check that clicking on a thread in the inbox list opens the thread with:
   1. the messages passing the current filter expanded (all for ALL MESSAGES, sent for SENT MESSAGES, and unread for UNREAD MESSAGES)
   1. the active message being the first message that passes the filter
1. Check that messages that start as collapsed expand on click, and stay expanded after clicking another message
1. Check that for each filter, the inbox shows thread that contain one or more messages of the filter type.

#### Any background context you want to provide?
This branch sits on the #235 branch.

#### Screenshots (if appropriate):
